File: python/src/agent_api/types/memories.py
from __future__ import annotations

from typing import Any, Literal, TypedDict

from agent_api.types.common import Usage


class MemorySearchParams(TypedDict, total=False):
    query: str
    limit: int
    previous_response_id: str
    tenant_search: bool
    lang: str
    semantic_weight: float


class MemorySearchHit(TypedDict, total=False):
    id: str
    score: float
    thread_id: str
    created_at: int
    fact: str
    tenant_id: str
    user_id: str
    response_id: str
    metadata: Any
    metadata_text: str


class MemorySearchModelUsage(TypedDict, total=False):
    source: str
    phase: str
    provider: str
    model: str
    attempt_index: int
    status: str
    usage: Usage


class MemorySearchResponse(TypedDict, total=False):
    object: Literal["memory_search_result"]
    data: list[MemorySearchHit]
    total: int
    rewritten_query: str
    model_usage: list[MemorySearchModelUsage]

File: javascript/src/types/memories.ts
import type { Usage } from "./common.js";

export interface MemorySearchParams {
  query: string;
  limit?: number;
  previous_response_id?: string;
  tenant_search?: boolean;
  lang?: string;
  semantic_weight?: number;
}

export interface MemorySearchHit {
  id: string;
  score?: number;
  thread_id?: string;
  created_at?: number;
  fact: string;
  tenant_id?: string;
  user_id?: string;
  response_id?: string;
  metadata?: unknown;
  metadata_text?: string;
}

export interface MemorySearchModelUsage {
  source?: string;
  phase?: string;
  provider?: string;
  model?: string;
  attempt_index?: number;
  status?: string;
  usage?: Usage;
}

export interface MemorySearchResponse {
  object: "memory_search_result";
  data: MemorySearchHit[];
  total?: number;
  rewritten_query?: string;
  model_usage?: MemorySearchModelUsage[];
}
